lib/upgrade.js

```diff
@@ -15,10 +15,10 @@ async function getAppDependenciesUtil(packageJson) {
 
   const majorVersion = builderVersionInstalled.split('.')[0];
   switch (majorVersion) {
-    case '4':
-      return require('./app-dependencies');
-    default:
+    case '3':
```

Comment:
Reversed the `switch` statement's logic so that it prefers the new utility, and only calls the legacy utility if the version is exactly `3`.